saleor/graphql/core/mutations.py

```diff
@@ -65,146 +47,159 @@ def _update_mutation_arguments_and_fields(cls, arguments, fields):
         cls._meta.fields.update(fields)
 
 
-class ModelFormMutationOptions(MutationOptions):
-    form_class = None
-    return_field_name = None
-
-
-class ModelDeleteMutationOptions(MutationOptions):
-    model = None
-    return_field_name = None
-
-
-class ModelDeleteMutation(BaseMutation):
-
+class ModelMutation(BaseMutation):
     class Meta:
         abstract = True
 
     @classmethod
     def __init_subclass_with_meta__(
-            cls, arguments=None, model=None, return_field_name=None,
-            _meta=None, **options):
+            cls, arguments=None, model=None, permissions=None, _meta=None,
+            **options):
         if not model:
-            raise ImproperlyConfigured(
-                'model is required for ModelDeleteMutation')
-
-        _meta = ModelDeleteMutationOptions(cls)
-
-        if not return_field_name:
-            return_field_name = get_model_name(model)
+            raise ImproperlyConfigured('model is required for ModelMutation')
+        _meta = ModelMutationOptions(cls)
```

Comment:
Do we have to provide `_meta=None` argument if it is being overwritten here anyway?